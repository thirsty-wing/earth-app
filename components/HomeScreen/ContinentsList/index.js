import ContinentsList from "./ContinentsList.datalayer";

export default ContinentsList;
