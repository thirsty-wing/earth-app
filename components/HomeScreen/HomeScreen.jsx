import { Text, View } from "react-native";
import { useSafeAreaInsets } from "react-native-safe-area-context";

function HomeScreen() {
  const insets = useSafeAreaInsets();

  return(
    <View
      style={{
        paddingBottom: insets.bottom,
        paddingLeft: insets.left,
        paddingRight: insets.right,
        paddingTop: insets.top,
        flex: 1,
      }}
    >
      <Text>Hello World!</Text>
    </View>
  );
}

export default HomeScreen;
