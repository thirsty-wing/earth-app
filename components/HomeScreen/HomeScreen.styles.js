import { StyleSheet } from "react-native";
import { theme } from "../../utils/theme";


export const styles = StyleSheet.create({
  titleView: {
    paddingHorizontal: theme.spacing(3),
    paddingBottom: theme.spacing(),
  },
  continentsListView: {
    gap: theme.spacing(),
    paddingHorizontal: theme.spacing(3),
  }
});
