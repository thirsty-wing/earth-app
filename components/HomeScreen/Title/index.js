import Title from "./Title.datalayer";

export default Title;
