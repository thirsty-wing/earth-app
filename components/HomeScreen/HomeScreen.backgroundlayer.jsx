import { View } from "react-native";

import HomeScreen from "./HomeScreen";

function HomeScreenBackgroundlayer() {
  return(
    <View
      style={{
        position: "absolute",
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
      }}
    >
      <HomeScreen/>
    </View>
  );
}

export default HomeScreenBackgroundlayer;
