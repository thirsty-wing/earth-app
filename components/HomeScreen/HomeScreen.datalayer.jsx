import { SafeAreaProvider } from "react-native-safe-area-context";

import HomeScreen from "./HomeScreen";

function HomeScreenDatalayer() {
  return(
    <SafeAreaProvider>
      <HomeScreen/>
    </SafeAreaProvider>
  );
}

export default HomeScreenDatalayer;
