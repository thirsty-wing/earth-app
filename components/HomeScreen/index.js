import HomeScreen from "./HomeScreen.datalayer";

export default HomeScreen;
