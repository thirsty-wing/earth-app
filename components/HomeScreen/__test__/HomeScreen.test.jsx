//import safeAreaContextMock from "react-native-safe-area-context/jest/mock";

import HomeScreen from "../HomeScreen";
import { shallow } from "../../../utils/test/shallow";


jest.mock("react-native-safe-area-context", () => ({
    useSafeAreaInsets: () => ({
      left: 0,
      right: 0,
      top: 0,
      bottom: 0,
    }),
  })
);


it("should render", () => {
  const content = shallow(<HomeScreen/>);

  expect(content).toMatchSnapshot();
});

