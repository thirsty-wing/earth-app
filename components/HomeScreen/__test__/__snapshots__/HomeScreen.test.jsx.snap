// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<BackgroundView>
  <View
    style={
      {
        "paddingHorizontal": 24,
      }
    }
  >
    <TitleDatalayer />
  </View>
  <ScrollView
    style={
      {
        "gap": 8,
        "paddingHorizontal": 24,
      }
    }
  >
    <ContinentsListDatalayer />
  </ScrollView>
</BackgroundView>
`;
